pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -525,7 +525,7 @@ def save_function(self, value: Optional[Callable]) -> None:
 
     def _del_model(self, trainer: "pl.Trainer", filepath: str) -> None:
         if trainer.should_rank_save_checkpoint and self._fs.exists(filepath):
-            self._fs.rm(filepath)
+            self._fs.rm(filepath, recursive=True)
```

Comment:
Need to add a test for model checkpoint when using deepspeed. Since we're saving a model as a directory in DeepSpeed, needs to be recursive

Reply:
Sean, is there a test for this?

Reply:
It's covered in the deepspeed save/load here: https://github.com/PyTorchLightning/pytorch-lightning/blob/fix/ds_saving_2/tests/plugins/test_deepspeed_plugin.py#L576